src/encoded/static/components/filegallery.js

```diff
@@ -2525,38 +2528,36 @@ AnalysesSelector.propTypes = {
 
 
 const TabPanelFacets = (props) => {
-    const { open, currentTab, filters, allFiles, filterFiles, toggleFacets, clearFileFilters, experimentType, analyses, selectedAnalysesIndex, handleAnalysesSelection, analysisSelectorRef } = props;
-
-    // Filter file list to make sure it includes only files that should be displayed
-    let fileList = allFiles;
-    if (currentTab === 'browser') {
-        fileList = filterForVisualizableFiles(fileList);
-    }
+    const { open, currentTab, filters, allFiles, filterFiles, toggleFacets, clearFileFilters, experimentType, analyses, selectedAnalysesIndex, handleAnalysesSelection, analysisSelectorRef, context } = props;
 
     // Initialize assembly object
     const assembly = { 'All assemblies': 100 };
 
-    // Create object for Assembly facet from list of all files
-    // We do not count how many results there are for a given assembly because we will not display the bars
-    fileList.forEach((file) => {
-        if (file.genome_annotation && file.assembly && !assembly[`${file.assembly} ${file.genome_annotation}`]) {
-            assembly[`${file.assembly} ${file.genome_annotation}`] = computeAssemblyAnnotationValue(file.assembly, file.genome_annotation);
-        } else if (file.assembly && !assembly[file.assembly] && !(file.genome_annotation)) {
-            assembly[file.assembly] = computeAssemblyAnnotationValue(file.assembly);
-        }
-    });
+    // set up filesFilteredByAssembly initially with files from drop down
+    const compiledAnalysis = compileAnalyses(context, allFiles);
+    const dropdown = analysisSelectorRef.current;
+    const selectedIndex = dropdown ? dropdown.selectedIndex : null;
+    const analysis = compiledAnalysis[selectedIndex];
+    const compileAnalysisFiles = analysis ? analysis.files : null;
+    let files = compileAnalysisFiles
+        ? allFiles.filter((f) => compileAnalysisFiles && compileAnalysisFiles.includes(f['@id']))
+        : allFiles;
+
+    if (currentTab === 'browser') {
+        files = filterForVisualizableFiles(files);
+    }
```

Comment:
It’s OK, but it seems like a shame to possibly filter `allFiles` and assign to `files`, just to throw that away if the browser tab is current. `allFiles` can potentially contain hundreds of thousands of files.